pkg/profiles/openapi.go

```diff
@@ -71,31 +75,38 @@ func swaggerToServiceProfile(swagger spec.Swagger, namespace, name, clusterDomai
 		path := path.Join(swagger.BasePath, relPath)
 		pathRegex := pathToRegex(path)
 		if item.Delete != nil {
-			spec := mkRouteSpec(path, pathRegex, http.MethodDelete, item.Delete.Responses)
+			retryable, _ := item.Delete.VendorExtensible.Extensions.GetBool("x-linkerd-retryable")
+			spec := mkRouteSpec(path, pathRegex, http.MethodGet, item.Delete.Responses, retryable)
```

Comment:
I think you can clean up some of the repetition by having `mkRouteSpec` accept a `spec.Exceptions` and put the code for extracting the retryable boolean in `mkRouteSpec`.

Reply:
@adleong Can you check now? I have tried to do some cleanup.